resources/migrations/001_update_migrations.yaml

```diff
@@ -11254,6 +11254,71 @@ databaseChangeLog:
                 'type/Number'
               )
 
+  - changeSet:
+      id: v55.2025-05-16T01:00:00
+      validCheckSum: ANY
+      author: ranquild
+      comment: Deduplicate metabase_database on (name, engine)
+      preConditions:
+        - onFail: MARK_RAN
+        - sqlCheck:
+            expectedResult: 1
+            sql: >-
+              SELECT CASE WHEN COUNT(*) > 0 THEN 1 ELSE 0 END
+              FROM (
+                SELECT ROW_NUMBER() OVER (PARTITION BY name, engine) AS row_num
+                FROM metabase_database
+              ) AS d
+              WHERE d.row_num > 1;
+      changes:
+        - sql:
+            dbms: postgresql
+            sql: >-
+              UPDATE metabase_database AS d1
+              SET name = CONCAT(d1.name, ' ', d2.row_num)
+              FROM (
+                SELECT id, ROW_NUMBER() OVER (PARTITION BY name, engine ORDER BY created_at) AS row_num
+                FROM metabase_database
+              ) AS d2
+              WHERE d1.id = d2.id AND d2.row_num > 1;
+        - sql:
+            dbms: mysql,mariadb
+            sql: >-
+              UPDATE metabase_database AS d1
+              INNER JOIN (
+                SELECT id, ROW_NUMBER() OVER (PARTITION BY name, engine ORDER BY created_at) AS row_num
+                FROM metabase_database
+              ) AS d2 ON d1.id = d2.id
+              SET name = CONCAT(d1.name, ' ', d2.row_num)
+              WHERE d2.row_num > 1;
+        - sql:
+            dbms: h2
+            sql: >-
+              UPDATE metabase_database AS d1
+              SET name = (
+                SELECT CASE WHEN d2.row_num > 1 THEN CONCAT(d1.name, ' ', d2.row_num) ELSE d1.name END
+                FROM (
+                  SELECT id, ROW_NUMBER() OVER (PARTITION BY name, engine ORDER BY created_at) AS row_num
+                  FROM metabase_database
+                ) AS d2
+                WHERE d1.id = d2.id
+              );
+      rollback: # cannot roll back
```

Comment:
This is a big red flag.

I think we should have a migration before this one that adds a column like `name_before_deduplication`. Then the rollback for this changeset is to `UPDATE metabase_database SET name = name_before_deduplication`.

I know it sucks, but downgrades happen.

Reply:
Agree, fixed!